src/widgets/scale/lv_scale.c

```diff
@@ -101,20 +139,106 @@ static void lv_scale_event(const lv_obj_class_t * class_p, lv_event_t * event)
+    lv_event_code_t event_code = lv_event_get_code(event);
     lv_obj_t * obj = lv_event_get_target(event);
     lv_scale_t * scale = (lv_scale_t *)lv_event_get_target(event);
+
+    if(event_code == LV_EVENT_DRAW_MAIN) {
+        if (LV_SCALE_MODE_HORIZONTAL == scale->mode) {
+            draw_horizontal_scale(obj, event);
+        }
+        else if (LV_SCALE_MODE_VERTICAL == scale->mode) {
+
+        }
+        else if (LV_SCALE_MODE_ROUND == scale->mode) {
+            
+        }
+        else {
+            /* Invalid mode */
+        }
+    }
+    else {
+        /* TODO */
+    }
 }
 
-static void lv_scale_draw(lv_event_t * event)
+static void draw_horizontal_scale(lv_obj_t *obj, lv_event_t * event)
 {
-    lv_obj_t * obj = lv_event_get_target(event);
     lv_scale_t * scale = (lv_scale_t *)obj;
-
     lv_draw_ctx_t * draw_ctx = lv_event_get_draw_ctx(event);
 
+    lv_point_t point1;
+    lv_point_t point2;
+
+    lv_draw_label_dsc_t label_dsc;
+    lv_draw_label_dsc_init(&label_dsc);
+    lv_obj_init_draw_label_dsc(obj, LV_PART_TICKS, &label_dsc);
+
+    lv_draw_line_dsc_t line_dsc;
+    lv_draw_line_dsc_init(&line_dsc);
+    lv_obj_init_draw_line_dsc(obj, LV_PART_TICKS, &line_dsc);
+
+    lv_coord_t pad_left = lv_obj_get_style_pad_left(obj, LV_PART_MAIN) + lv_obj_get_style_border_width(obj, LV_PART_MAIN);
+    lv_coord_t w = (int32_t)lv_obj_get_content_width(obj);
+
+    lv_coord_t x_ofs = obj->coords.x1 + pad_left - lv_obj_get_scroll_left(obj);
+    lv_coord_t y_ofs = obj->coords.y2;
+    lv_coord_t label_gap;
+    label_gap = scale->label_enabled ? lv_obj_get_style_pad_bottom(obj, LV_PART_TICKS) : 0;
+
+    /* Line descriptor customization */
+    line_dsc.dash_gap = 0;
+    line_dsc.dash_width = 0;
+
     lv_obj_draw_part_dsc_t part_draw_dsc;
     lv_obj_draw_dsc_init(&part_draw_dsc, draw_ctx);
-}
+    part_draw_dsc.class_p = MY_CLASS;
+    part_draw_dsc.type = LV_SCALE_DRAW_PART_TICK_LABEL;
+    part_draw_dsc.id = LV_SCALE_MODE_HORIZONTAL;
+    part_draw_dsc.part = LV_PART_TICKS;
+    part_draw_dsc.label_dsc = &label_dsc;
+    part_draw_dsc.line_dsc = &line_dsc;
+
+    point1.y = y_ofs;
+    uint8_t tick_idx = 0;
+    for (tick_idx = 0; tick_idx < scale->total_tick_count; tick_idx++)
+    {
+        /*draw a line at moving x position*/
+        point2.x = point1.x = x_ofs + (int32_t)((int32_t)(w - line_dsc.width) * tick_idx) / 5;
+        point2.y = point1.y - 10; /* TODO */
+
+        LV_LOG_USER("Drawing tick %d at X: %d", tick_idx, point1.x);
+
+        part_draw_dsc.p1 = &point1;
+        part_draw_dsc.p2 = &point2;
+
+        part_draw_dsc.label_dsc = &label_dsc;
+        char text_buffer[20];
+        lv_snprintf(text_buffer, sizeof(text_buffer), "Hi %" LV_PRId32, tick_idx);
+        part_draw_dsc.label_dsc = &label_dsc;
+        part_draw_dsc.text = text_buffer;
+        part_draw_dsc.text_length = 10;
+
+        lv_event_send(obj, LV_EVENT_DRAW_PART_BEGIN, &part_draw_dsc);
```

Comment:
`LV_EVENT_DRAW_PART_*`  doesn't need to be added as they are already removed in `feat/parallel-rendering`.

Reply:
I'll add a TODO comment to remove them, if I remove it now I won't be able to continue as I'm using a bit old commit of lvgl.